XamlControlsGallery/Navigation/NavigationRootPage.xaml.cs

```diff
@@ -281,19 +282,34 @@ private void OnControlsSearchBoxQuerySubmitted(AutoSuggestBox sender, AutoSugges
 
         private void NavigationViewControl_PaneClosing(Microsoft.UI.Xaml.Controls.NavigationView sender, Microsoft.UI.Xaml.Controls.NavigationViewPaneClosingEventArgs args)
         {
-            AppTitle.Visibility = Visibility.Collapsed;
+            UpdateAppTitleMargin(sender);
         }
 
         private void NavigationViewControl_PaneOpened(Microsoft.UI.Xaml.Controls.NavigationView sender, object args)
         {
-            AppTitle.Visibility = Visibility.Visible;
-            if (sender.DisplayMode == Microsoft.UI.Xaml.Controls.NavigationViewDisplayMode.Expanded)
+            UpdateAppTitleMargin(sender);
+        }
+
+        private void NavigationViewControl_DisplayModeChanged(Microsoft.UI.Xaml.Controls.NavigationView sender, Microsoft.UI.Xaml.Controls.NavigationViewDisplayModeChangedEventArgs args)
+        {
+            UpdateAppTitleMargin(sender);
+        }
+
+        private void UpdateAppTitleMargin(Microsoft.UI.Xaml.Controls.NavigationView sender)
+        {
+            if (!(ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 7)))
+            {
+                AppTitle.TranslationTransition = new Vector3Transition();
+            }
+
+            if (sender.IsPaneOpen == false && (sender.DisplayMode == Microsoft.UI.Xaml.Controls.NavigationViewDisplayMode.Expanded ||
+                sender.DisplayMode == Microsoft.UI.Xaml.Controls.NavigationViewDisplayMode.Compact))
             {
-                AppTitleBar.Margin = new Thickness(40, 0, 0, 0);
+                AppTitle.Translation = new System.Numerics.Vector3(34, 0, 0);
```

Comment:
Similarly, this will cause crashes where the Translation property is not available. I think we'll need two codepaths: One that sets margin pre-RS5 and one that sets Translation post RS5.

Reply:
Yeah, I was thinking about Vector3 being available for a while but somehow failed to notice that Translation is new. Also fixed now.